application.py

```diff
@@ -183,15 +190,25 @@ def handleUSECluster2():
     if(len(data) < 6):
         raise BeagleError(errors.TOO_FEW_QUESTIONS)
     
-    embeddings, data_used_for_demo, q_ids_list = get_data_embeddings(data)
-    
-    best_scores = list(map(int,get_best_HAC_normal(embeddings, data_used_for_demo)[1]))
-    return jsonify(return_cluster_dict(best_scores,q_ids_list))
+    embeddings, questions_text, q_ids_list = get_data_embeddings(data)
```

Comment:
Given that you are now doing this calculation of taking an output, then calculating labels, then building an output, you should create a new function that does this calculation and simply call it in each of your three api routes.

Reply:
In fact, Jeff explicitly asked you to do this in another post and you handled the other two things he asked for but not this. Let's talk about this on our call.

Reply:
Hey Turner, I agree! Apologies for missing this out earlier. I have optimized this by creating a single function that can be called in the API routes. This saved both code lines and number of imports! Also increased readability! :D